### PR TITLE
core/chains/evm/config/toml/defaults: remove ineffectual overrides

### PR DESCRIPTION
**Pull request by jmank88**
This PR proposes removing redundant/ineffectual overrides from the evm configs. I left a few behind which had long comments about lack of support, so that we don't accidentally change them.

### CODE REVIEW

File: core/chains/evm/config/toml/defaults/Gnosis_Mainnet.toml [hunk 1]
```diff
@@ -13,7 +13,3 @@ LogPollInterval = '5s'
 [GasEstimator]
 PriceDefault = '1 gwei'
 PriceMax = '500 gwei'
-# 1 Gwei is the minimum accepted by the validators (unless whitelisted)
```
Comment (dimriou) [inline core/chains/evm/config/toml/defaults/Gnosis_Mainnet.toml:16]: I think this comment is valuable and we might want to keep it regardless of the config

Comment (jmank88) [reply core/chains/evm/config/toml/defaults/Gnosis_Mainnet.toml:16]: Ah I misread it as `wei` :facepalm:

Comment (dimriou) [reply core/chains/evm/config/toml/defaults/Gnosis_Mainnet.toml:16]: Fallback is 1 gwei so it can be removed, I was just referring to the comment.

**Review by dhaidashenko — APPROVED** [review]
LGTM.
`TestFunctionsConnectorHandler` fails, but it seems to be a flake not caused by the changes

**Review by dimriou — APPROVED** [review]

**Review by samsondav — APPROVED** [review]

File: core/chains/evm/config/toml/defaults/BSC_Mainnet.toml [hunk 2]
```diff
@@ -2,32 +2,19 @@
 # Clique offers finality within (N/2)+1 blocks where N is number of signers
 # There are 21 BSC validators so theoretically finality should occur after 21/2+1 = 11 blocks
 ChainID = '56'
-# Keeping this >> 11 because it's not expensive and gives us a safety margin
```
Comment (prashantkumar1982) [inline core/chains/evm/config/toml/defaults/BSC_Mainnet.toml:5]: Maybe this comment ought to remain, to show why we chose to go with eth's depth for BSC too?

**Review by prashantkumar1982 — COMMENTED** [review]
Looks good, just 1 minor comment

**Pull request merged**